src/dvc_objects/fs/system.py

```diff
@@ -91,7 +91,7 @@ def reflink(src, dst):
         src_fd = os.open(src, os.O_RDONLY)
 
         try:
-            dst_fd = os.open(dst, os.O_WRONLY | os.O_CREAT | os.O_TRUNC)
+            dst_fd = os.open(dst, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, 0o666)
```

Comment:
```suggestion
            dst_fd = os.open(dst, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, 0o666 & ~umask)
```
this would be the proper way, strictly speaking

Reply:
Isn't `os.open` doing that? Let me take a look.

Reply:
Docs says:

> Open the file path and set various flags according to flags and possibly its mode according to mode. When computing mode, the current umask value is first masked out

Reply:
My bad, mixed things up.